app/api/v1/dependencies/_domain.py

```
@@ -702,6 +862,93 @@ async def get_chain_anchor_repo(
+async def get_event_stream_broadcaster() -> InMemoryEventStreamBroadcaster:
+    """Return the in-memory SSE broadcaster instance (singleton)."""
+    # Event stream broadcaster is constructed in lifespan and stored on app.state;
+    # for now, we expose the global singleton as the broadcaster dependency.
+    # This can be swapped out in tests.
+    from app.infrastructure.services.event_stream_broadcaster import (
+        InMemoryEventStreamBroadcaster as _Broadcaster,
+    )
+
+    # In production, lifespan wires this; here we ensure a singleton for DI/tests.
+    if not hasattr(get_event_stream_broadcaster, "_instance"):
+        setattr(get_event_stream_broadcaster, "_instance", _Broadcaster())
```

Ruff (0.15.5)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)